crates/sui-graphql-rpc/src/types/big_int.rs

```diff
@@ -47,6 +47,13 @@ impl FromStr for BigInt {
     }
 }
 
+impl From<u64> for BigInt {
+    fn from(value: u64) -> Self {
+        // Number(value.into()) panics, so maybe this is not the right way to handle it
+        <BigInt as ScalarType>::parse(Value::String(value.to_string())).unwrap()
```

Comment:
how about BigInt(value.to_string())?

Reply:
Nvm, sounds like this would bypass the existing validation logic

Reply:
I think there is a valid comment here, namely that `BigInt` should have a constructor that does not require you to wrap the type in an `async-graphql` `Value` first.

Also, this is not introduced in your PR, but I just noticed that the `FromStr` implementation for `BigInt` is not correct -- it's infallible, and accepts any string without question, so you can use it to wrap arbitrary strings in `BigInt` which is pretty unfortunate.  Not something that you need to fix for this PR, but by fixing it, you could address this comment as well.

@stefan-mysten, here is what I suggest:

- Move the logic for parsing a `Value::String(s)` into a `BigInt` into its `FromStr` implementation (i.e. `from_str` should be checking that the string contains all digits and an optional leading `-`).
- Allow `FromStr` to return an error (there's only one error case -- `NotANumber`).
- Adapt `BigInt::parse` to use this new `from_str` implementation.
- Also use it in the `from` implementation.
- Replace the use of `unwrap` with an `expect(...)` in `BigInt::from(u64)`, so we get a nice message explaining what happened if that happens to trigger.

Reply:
@amnn - could you check now? Not sure I got the `from_str` stuff right.